### DISCUSSION TITLE
Exclude folders

### DISCUSSION THREAD

**Discussion by otaviosoato** [actions/cache#538] [Q&A]
Hi folks,

I don't know if I'm doing something wrong, but I can't exclude folder from cache using "!" before the path.

``` 
- uses: actions/cache@v2.1.4
  with:
    path: |
      Library/
      !Library/Artifacts/ 
```
The snippet above should work, right?

PS: English is not my native language. Sorry if I wrote something wrong.

Thank you!

**Comment by smorimoto** [comment]
Yes, this action internally uses the glob package of the actions toolkit, and it should support that.

https://github.com/actions/toolkit/tree/main/packages/glob

https://github.com/actions/toolkit/blob/383ec9fb0389e0473078971d5423323e79a575ff/packages/cache/src/internal/cacheUtils.ts#L42-L59
https://github.com/actions/toolkit/blob/383ec9fb0389e0473078971d5423323e79a575ff/packages/cache/src/cache.ts#L157

**Reply by dhadka** [reply]
Glob does support exclusions but unfortunately because the cache just matches the top-level folders and does not traverse any descendants, exclusions on nested folders don't work.  https://github.com/actions/toolkit/issues/713 has more details.

**Reply by otaviosoato** [reply]
Thanks @dhadka ! I'm got crazy trying to make that work. But I can rest in peace now haha
